src/tr2/game/savegame/savegame_bson.c

```diff
@@ -366,6 +368,9 @@ static bool M_LoadMusic(
         }
     }
 
+    current_track = M_ConvertMusicTrack(current_track, header_version);
+    ambient_track = M_ConvertMusicTrack(ambient_track, header_version);
```

Comment:
Let's include a comment that explains which game versions this fix is applicable for so that we can remove this once sufficient time has passed.